client/virtual_channel_test.go

```diff
@@ -228,3 +282,14 @@ func setupVirtualChannelTest(t *testing.T, ctx context.Context) (vct virtualChan
 	require.NoError(err, "updating virtual channel")
 	return
 }
+
+func (vct virtualChannelTest) registerAndSettle(ctx context.Context, ch *client.Channel) (err error) {
+	err = ch.Register(ctx)
+	if err != nil {
+		return
+	}
+
+	time.Sleep(100 * time.Millisecond) // Sleep to ensure that registered event has been processed.
```

Comment:
As we noticed in the intro discussing; this is not ideal.  
Some kind of wait semantic which waits for a specific state version is needed in the long run.

Reply:
Created issue #137